lib/Xmds/Soap4.php

```diff
@@ -271,7 +272,18 @@ public function RegisterDisplay($serverKey, $hardwareKey, $displayName, $clientT
         $display->clientType = $clientType;
         $display->clientVersion = $clientVersion;
         $display->clientCode = $clientCode;
-        //$display->operatingSystem = $operatingSystem;
+
+        // Parse operatingSystem data
+        $operatingSystem = json_decode($operatingSystem, false);
+
+        // Newer version of players will return a JSON value, but for older version, it will return a string.
+        // In case the json decode fails, use the operatingSystem string value as the default value for the osVersion.
+        $display->osVersion = isset($operatingSystem) ? $operatingSystem->osVersion : $operatingSystem;
```

Comment:
If json_decode fails, the operating system string value will be `false` (because the return of json_decode will be false overwriting the variable.

You'll probably need another temporary variable 🙂

Reply:
Thank you again for the review! Please see the latest changes [here](https://github.com/xibosignage/xibo-cms/pull/2574/commits/803b8ba158a782468b7a6cb41692978118c396fc).